tools/replace_code_block.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import sys
+from traceback import print_tb
+from typing import AnyStr
+
+status = 0
+
+
+def check_for_code(file: AnyStr, check: bool):
+    """Check a file and replace ``.. code::`` to ``.. code-block::``
+
+    :param file: The file to open
+    :param check: Whether or not the program is operating in ``check`` mode.
+        In check mode, the program will exit with a status 1 if files will be
+        replaced. In non-check mode, the program will replace files.
+    """
+    global status
+    with open(file, "rb") as infile:
+        content = infile.read()
+    old_content = content
+    new_content = content.replace(b".. code::", b".. code-block::")
+    if old_content != new_content:
+        if check:
+            status = 1
+            print(
+                "File {filename} contains ``code::`` statements!".format(
+                    filename=file
+                )
+            )
+        else:
+            with open(file, "wb") as outfile:
+                outfile.write(new_content)
+            print(
+                "Replaced ``code::`` statements to ``code-block::`` in "
+                "{filename}".format(filename=file)
+            )
+
+
+def main(check):
+    global status
+    for current_directory, folder_list, file_list in os.walk(
+        os.path.abspath(os.path.join(__file__, "..", "..", "praw"))
```

Comment:
Is `abspath` necessary here?

Reply:
Without it, the ".." is included in the path, so abspath gets rid of those

Reply:
Is it a problem if `..` is in the path?

Reply:
I don't know if it would be a problem but python could do weird things processing code like that, and it's not like the abspath is a significant overhead